codegen-core/common-test-models/pokemon.smithy

```diff
@@ -14,135 +14,131 @@ use com.aws.example#ResourceNotFoundException
-        GetServerStatistics,
-        DoNothing,
-        CapturePokemon,
-        CheckHealth,
+        GetServerStatistics
+        DoNothing
+        CapturePokemon
+        CheckHealth
         StreamPokemonRadio
-    ],
+    ]
 }
 
 /// A users current Pokémon storage.
 resource Storage {
     identifiers: {
         user: String
-    },
-    read: GetStorage,
+    }
+    read: GetStorage
 }
 
 /// Retrieve information about your Pokédex.
 @readonly
 @http(uri: "/pokedex/{user}", method: "GET")
 operation GetStorage {
-    input: GetStorageInput,
-    output: GetStorageOutput,
-    errors: [ResourceNotFoundException, StorageAccessNotAuthorized, ValidationException],
+    input := @sensitive @documentation("A request to access Pokémon storage.") {
+        @required
+        @httpLabel
+        user: String
+        @required
+        @httpHeader("passcode")
+        passcode: String
+    }
+    output := @documentation("Contents of the Pokémon storage.") {
+        @required
+        collection: SpeciesCollection
+    }
+    errors: [
+        ResourceNotFoundException
+        StorageAccessNotAuthorized
+        ValidationException
+    ]
 }
 
 /// Not authorized to access Pokémon storage.
 @error("client")
 @httpError(401)
 structure StorageAccessNotAuthorized {}
 
-/// A request to access Pokémon storage.
-@input
-@sensitive
-structure GetStorageInput {
-    @required
-    @httpLabel
-    user: String,
-    @required
-    @httpHeader("passcode")
-    passcode: String,
-}
-
 /// A list of Pokémon species.
 list SpeciesCollection {
     member: String
 }
 
-/// Contents of the Pokémon storage.
-@output
-structure GetStorageOutput {
-    @required
-    collection: SpeciesCollection
-}
-
 /// Capture Pokémons via event streams.
 @http(uri: "/capture-pokemon-event/{region}", method: "POST")
 operation CapturePokemon {
-    input: CapturePokemonEventsInput,
-    output: CapturePokemonEventsOutput,
-    errors: [UnsupportedRegionError, ThrottlingError, ValidationException]
-}
-
-@input
-structure CapturePokemonEventsInput {
-    @httpPayload
-    events: AttemptCapturingPokemonEvent,
-
-    @httpLabel
-    @required
-    region: String,
-}
-
-@output
-structure CapturePokemonEventsOutput {
-    @httpPayload
-    events: CapturePokemonEvents,
+    input := {
+        @httpPayload
+        events: AttemptCapturingPokemonEvent
+
+        @httpLabel
+        @required
+        region: String
+    }
+    output := {
+        @httpPayload
+        events: CapturePokemonEvents
+    }
+    errors: [
+        UnsupportedRegionError
+        ThrottlingError
+        ValidationException
+    ]
 }
 
 @streaming
 union AttemptCapturingPokemonEvent {
-    event: CapturingEvent,
-    masterball_unsuccessful: MasterBallUnsuccessful,
+    event: CapturingEvent
+    masterball_unsuccessful: MasterBallUnsuccessful
 }
 
 structure CapturingEvent {
     @eventPayload
-    payload: CapturingPayload,
+    payload: CapturingPayload
 }
 
 structure CapturingPayload {
```

Comment:
nit: Should we use [target elision syntax](https://smithy.io/2.0/guides/model-translations/migrating-idl-1-to-2.html#use-the-target-elision-syntax-sugar-to-reduce-boilerplate) to clarify that the `name` refers to pokemon's name?

Reply:
It felt like the feature complicated the model, but since it is an example/sample test, probably good to use as many features as possible. I'll get this updated